x/global/manager.go

```diff
@@ -529,94 +532,134 @@ func (gm *GlobalManager) ClearEventCache(ctx sdk.Context) sdk.Error {
 	return nil
 }
 
+func (gm *GlobalManager) ExportToFile(ctx sdk.Context, cdc *codec.Codec, filepath string) error {
+	state := &model.GlobalTablesIR{
+		Version: exportVersion,
+	}
+	storeMap := gm.storage.PartialStoreMap(ctx)
+
+	// export events
+	storeMap[string(model.TimeEventListSubStore)].Iterate(func(key []byte, val interface{}) bool {
+		ts, err := strconv.ParseInt(string(key), 10, 64)
+		if err != nil {
+			panic(err)
+		}
+		events := val.(*types.TimeEventList)
+		state.GlobalTimeEventLists = append(state.GlobalTimeEventLists, model.GlobalTimeEventsIR{
```

Comment:
All coin return event can be omitted and record the amount for each user.

Reply:
not part of this pr. will add later for this special case.